servers/store/src/main/java/org/projectnessie/server/store/IcebergTableSerializer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.store;
+
+import com.google.protobuf.ByteString;
+import java.util.function.Supplier;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.server.store.proto.ObjectTypes;
+
+public final class IcebergTableSerializer extends BaseSerializer<IcebergTable> {
+
+  @Override
+  public String contentType() {
+    return Content.Type.ICEBERG_TABLE.name();
+  }
+
+  @Override
+  public Content.Type getType(byte payload, ByteString onReferenceValue) {
+    return Content.Type.ICEBERG_TABLE;
```

Comment:
What is the purpose of the args now? If only for test, could we validate payload to value match some other way, without overloading the main production call path with extra args?

Reply:
Ah, thx, I missed that... However, would it make sense to rename `UnknownSerializer` to `LegacySerializer` perhaps?.. just for the sake of clarity... I'd even call it `LegacyDeerializer`.

Reply:
The type‘s named `UNKNOWN`, that’s why

Reply:
But them we have type misalignment between `contentType()` and `getType(byte payload, ByteString onReferenceValue)`.

`contentType()` appears to be required only for serialization, which is not supported by `UnknownSerializer`... looks like of odd.

What do you think about making the `0` payload handling a special case in `DefaultStoreWorker`? Then, we may not need `UnknownSerializer` at all :thinking:

Reply:
`DefaultStoreWorker` shouldn't know anything about the "main" content types.
We can likely remove the functionality in `UnknownSerializer` when there are no more commits with payload==0 - the state where we'll eventually be.

Reply:
I.e. `UnknownSerializer` doesn't need to be perfect - it'll go away.

Reply:
... and the `getType()` function will then disappear as well

Reply:
yes, but it still looks awkward. How about using a separate type for `0` payload, i.e. `LEGACY`, which would _not_ have a constant in `Content.Type`. The `UNKNOWN` type's serialize would throw exceptions on ser/deser attempts and the `LEGACY` serializer would be used only for reading old entries with `0` payload. WDYT?

Reply:
Actually, we can probably resolve this after #4945

Reply:
`UNKNOWN` itself is useful - i.e. when there's a payload for which you no longer have a serializer/content.

it's just that accidentally existing persisted data has payload==0, which should have never happened, but probably did.

Reply:
Actually, we can probably resolve this after #4945